codex/rest/api.nim

```diff
@@ -52,88 +50,87 @@ proc validate(
-    "addresses": jarray
+    "content": jarray
   }
   return jobj
 
-proc initRestApi*(node: CodexNodeRef, conf: CodexConf): RestRouter =
-  var router = RestRouter.init(validate)
-  router.api(
-    MethodGet,
-    "/api/codex/v1/connect/{peerId}") do (
-      peerId: PeerId,
-      addrs: seq[MultiAddress]) -> RestApiResponse:
-      ## Connect to a peer
-      ##
-      ## If `addrs` param is supplied, it will be used to
-      ## dial the peer, otherwise the `peerId` is used
-      ## to invoke peer discovery, if it succeeds
-      ## the returned addresses will be used to dial
-      ##
-      ## `addrs` the listening addresses of the peers to dial, eg the one specified with `--listen-addrs`
+proc initContentApi(node: CodexNodeRef, router: var RestRouter) =
+  router.rawApi(
+    MethodPost,
+    "/api/codex/v1/content") do (
+    ) -> RestApiResponse:
+      ## Upload a file in a streaming manner
       ##
 
-      if peerId.isErr:
-        return RestApiResponse.error(
-          Http400,
-          $peerId.error())
+      trace "Handling file upload"
+      var bodyReader = request.getBodyReader()
+      if bodyReader.isErr():
+        return RestApiResponse.error(Http500)
+
+      # Attempt to handle `Expect` header
+      # some clients (curl), wait 1000ms
+      # before giving up
+      #
+      await request.handleExpect()
+
+      let
+        reader = bodyReader.get()
 
-      let addresses = if addrs.isOk and addrs.get().len > 0:
-            addrs.get()
-          else:
-            without peerRecord =? (await node.findPeer(peerId.get())):
-              return RestApiResponse.error(
-                Http400,
-                "Unable to find Peer!")
-            peerRecord.addresses.mapIt(it.address)
       try:
-        await node.connect(peerId.get(), addresses)
-        return RestApiResponse.response("Successfully connected to peer")
-      except DialFailedError:
-        return RestApiResponse.error(Http400, "Unable to dial peer")
-      except CatchableError:
-        return RestApiResponse.error(Http400, "Unknown error dialling peer")
+        without cid =? (
+          await node.store(AsyncStreamWrapper.new(reader = AsyncStreamReader(reader)))), error:
+          trace "Error uploading file", exc = error.msg
+          return RestApiResponse.error(Http500, error.msg)
+
+        codexApiUploads.inc()
+        trace "Uploaded file", cid
+        return RestApiResponse.response($cid)
+      except CancelledError:
+        trace "Upload cancelled error"
+        return RestApiResponse.error(Http500)
+      except AsyncStreamError:
+        trace "Async stream error"
+        return RestApiResponse.error(Http500)
+      finally:
+        await reader.closeWait()
+
+      trace "Something went wrong error"
+      return RestApiResponse.error(Http500)
+
+  router.api(
+    MethodGet,
+    "/api/codex/v1/content") do () -> RestApiResponse:
+      without cids =? await node.blockStore.listBlocks(BlockType.Manifest):
```

Comment:
we might want to expose a specific call in the `node` object for this, it's not a big deal, but the store was never meant to be accessed by this top layers.

Reply:
Excellent point. I did this, and I think it's a great improvement.